internal/host/host.go

```diff
@@ -1009,7 +1031,7 @@ func (m *Manager) AutoAssignRole(ctx context.Context, h *models.Host, db *gorm.D
 	if h.Role == models.HostRoleAutoAssign {
 		log := logutil.FromContext(ctx, m.log)
 		// If role is auto-assigned calculate the suggested roles
-		// This logic will moved to the monitor soon
+		// to make sure the suggestion is fresh
```

Comment:
In this case we calculate it one more time before installation, if the calculation was missed in the monitor. Here we do not need the AUTO_ASSIGN flag because it is the legacy place and code for such calculation. When the flag is set to false, there will be no calculation in the monitoring and only here we will decide on the actual role